File: Server/Models/Movie.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Models
{
	[Table("Movie")]
	public class Movie
	{
		[Key]
		public int ID { get; set; }

		[Required]
		[MaxLength(60)]
		public string Name { get; set; }
		
		[Required]
		[MaxLength(300)]
		public string PlotSummary { get; set; }

		[Required]
		public string PosterPath { get; set; }
		
		[Required]
		public int Length { get; set; }	
		
		[Required]
		public int Year { get; set; }
		
		[Required]
		[MaxLength(50)]
		public string Country { get; set; }
		
		[Required]
		[MaxLength(50)]
		public string Director { get; set; }

		public List<Genre> Genres { get; set; }

		public List<Screening> Screenings { get; set; }
	}
}

File: Client/scripts/models/movie.js
import { Screening } from "./screening.js";

export class Movie {
	constructor(id, name, posterPath, year) {
		this.id = id;
		this.name = name;
		this.posterPath = posterPath;
		this.year = year;
		this.screenings = [];
		this.summary = null;
		this.length = null;
		this.country = null;
		this.director = null;
		this.genres = [];
	}

	async getInfo() {
		if (this.summary !== null) {
			return;
		}

		try {
			const response = await fetch(`https://localhost:5001/Movie/GetInfo/${this.id}`);
		
			if (!response.ok) {
				throw new error("Response not OK.");
			}

			const data = await response.json();

			this.summary = data.plotSummary;
			this.length = data.length;
			this.country = data.country;
			this.director = data.director;
			this.genres = data.genres;
		} catch (error) {
			console.error(error);
		}
	}

	async getScreenings(cinemaId) {
		if (this.screenings.length != 0) {
			return;
		}

		try {
			const response = await fetch(`https://localhost:5001/Screening/GetScreeningsForMovie/${cinemaId}/${this.id}`);
		
			if (!response.ok) {
				throw new error(response.statusText);
			}

			const data = await response.json();

			for (const el of data) {
				const screening = new Screening(el.id, cinemaId, this.id, el.date, el.time, el.ticketPrice);
				this.screenings.push(screening);
			}
		} catch (error) {
			console.log(error);
		}
	}


}